src/bundles/create-async-resource-bundle.js

```diff
@@ -11,8 +9,24 @@ const defaultOpts = {
   persist: true
 }
 
+const requiredOpts = {
+  name: null,
+  getPromise: null
+}
+
 export default spec => {
-  const opts = Object.assign({}, defaultOpts, spec)
+  const opts = Object.assign({}, requiredOpts, defaultOpts, spec)
+
+  if (process.env.NODE_ENV !== 'production') {
+    for (const item in requiredOpts) {
+      if (opts[item] === null) {
```

Comment:
You know as I'm looking at this let's not merge in `requiredOpts` since they're required they'll be overwritten anyway. There's no reason to include them in defaultOpts, even.

```js
const defaultOpts = {
  staleAfter: 900000, // fifteen minutes
  retryAfter: 60000, //
  expireAfter: Infinity,
  checkIfOnline: true,
  persist: true
}
```

then we can track required options inside the env check too and dump more code in prod.

```js
const opts = Object.assign({}, defaultOpts, spec)

if (process.env.NODE_ENV !== 'production') {
  const requiredOptions = ['name', 'getPromise']  
  requiredOptions.forEach(item => {
    if (!opts[item]) {
      throw Error(`You must supply an ${item} option when creating a resource bundle`)
    }
  })
}
```

Reply:
Nice catch - can't believe I missed that.